test/functional/wallet_fundrawtransaction.py

```diff
@@ -596,49 +605,57 @@ def test_locked_wallet(self):
                 'active': True,
                 'internal': True
             }])
-            self.nodes[1].walletlock()
+            wallet.walletlock()
 
         # Drain the keypool.
-        self.nodes[1].getnewaddress()
-        self.nodes[1].getrawchangeaddress()
+        wallet.getnewaddress()
+        wallet.getrawchangeaddress()
 
-        # Choose 2 inputs
-        inputs = self.nodes[1].listunspent()[0:2]
-        value = sum(inp["amount"] for inp in inputs) - Decimal("0.00000500") # Pay a 500 sat fee
+        # Choose input
+        inputs = wallet.listunspent()
+        # Deduce fee to produce a changeless transaction
+        value = inputs[0]["amount"] - Decimal("0.00002200")
         outputs = {self.nodes[0].getnewaddress():value}
-        rawtx = self.nodes[1].createrawtransaction(inputs, outputs)
+        rawtx = wallet.createrawtransaction(inputs, outputs)
         # fund a transaction that does not require a new key for the change output
-        self.nodes[1].fundrawtransaction(rawtx)
+        funded_tx = wallet.fundrawtransaction(rawtx)
```

Comment:
Haven't been able to reproduce it. It is always passing here. But, I imagine that it's related to the dynamic fee rate calculation. Which.. is not needed in this test case; the test exercises the creation of a changeless transaction, the fee rate used is not really important.
So, when you can, try https://github.com/furszy/bitcoin-core/commit/557c50deb0253af299dc88856b34399b0d626bd3.

Reply:
> I wonder why implicit fee estimation passed in the first place.  Previously there were two inputs and a fee of 550sat. Now there is a fee of 2200 sat with one input.

That is because the default wallets have a custom tx fee (set [here](https://github.com/bitcoin/bitcoin/blob/b7138252ace6d21476964774e094ed1143cd7a1c/test/functional/wallet_fundrawtransaction.py#L100)), which is much lower than the one used in the wallet created by this test case (which runs the default fee estimation).

Side note; the implicit fee estimation suffered other adaptations already:

* https://github.com/bitcoin/bitcoin/commit/a165bfbe44b4db3a40b8d1ba8095035367c932a7 from [#17526](https://github.com/bitcoin/bitcoin/pull/17526).
* https://github.com/bitcoin/bitcoin/commit/86beee05795216738f51fa744539336503c26fd9 from [#22009](https://github.com/bitcoin/bitcoin/pull/22009).
* https://github.com/bitcoin/bitcoin/commit/92bcd70808b9cac56b184903aa6d37baf9641b37 from [#17219](https://github.com/bitcoin/bitcoin/pull/17219).

So.. seems that this number has been adapted every now and then. And wouldn't be bad to provide a specific fee rate to `fundrawtransaction()` and document the expected behavior as I did in https://github.com/furszy/bitcoin-core/commit/557c50deb0253af299dc88856b34399b0d626bd3.

> Mind creating a pull, as I can't test intermittent issues? I think this needs clarification/fixup even absent an intermittent issue.

Sure. Let me know if we are sync and will create the PR.